src/resources/token.js

```diff
@@ -16,6 +16,10 @@ class Token extends ResourceBase {
     this.getTokenAddress = async function() {
       return await marketplace.getTokenAddress()
     }
+
+    this.contractService = contractService
+    this.marketplace = marketplace
+    this.contractName = 'OriginToken'
```

Comment:
As opposed to hardcoding it, I think it is accessible via OriginTokenContract.contractName